service-runner.js

```diff
@@ -178,19 +204,35 @@ ServiceRunner.prototype._runMaster = function() {
     // Set up rolling restarts
     process.on('SIGHUP', this._rollingRestart.bind(this));
 
-    return this._startWorkers(this.config.num_workers);
+    return this._startWorkers(this.config.num_workers)
+    .then(function(workers) {
+        setTimeout(function() {
```

Comment:
@d00rman Unfortunately we don't control when the worker keys are added, as we are iterating over node's internal cluster.workers collections to check each worker. Also, if we add a worker to checks only when it sends a `startup_finished`, we may never start checking a worker, which died even before sending that message. I'll try to make it start sending beats a bit earlier in the startup chain, that should help.